src/Nethermind/Nethermind.Core/Buffers/ArrayPoolDisposableReturn.cs

```diff
@@ -0,0 +1,22 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Buffers;
+
+namespace Nethermind.Core.Buffers;
+
+public readonly struct ArrayPoolDisposableReturn : IDisposable
```

Comment:
why not use ArrayPoolList?

Reply:
Just wanted type so I could do 

```
using v = ;
```
As single line without either adding a `ArrayPool<byte>.Shared.Return` before each return; or going full try/finally and changing the indentation 😅

Reply:
ArrayPoolList has that!